src/AppTokenRetrival.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react'
+
+import { Mount } from 'react-lifecycle-components'
+import { useLocation } from 'react-router'
+
+import { apolloClient } from './client/apolloClient'
+import { StorageContainer } from './utils/StorageContainer'
+
+export const AppTokenRetrival = ({ children }) => {
+  const location = useLocation()
+
+  return (
+    <StorageContainer>
+      {(storageState) => (
+        <Mount
+          on={() => {
+            if (location.hash.includes('token')) {
+              var params = {}
+
+              location.hash
+                .substring(1)
+                .split('&')
+                .map((hk) => {
+                  let param = hk.split('=')
+                  params[param[0]] = param[1]
+                })
```

Comment:
This is a bit nit but using `Array#reduce` would be prettier ;) but i guess this is _acceptable_ although i think `forEach` would be more appropriate if we wouldn't want to use reduce for some reason

Reply:
hahah was to quick copying from stackoverflow, this is now a reduce hehe